src/solvers/fdsbp_unstructured/containers_2d.jl

```diff
@@ -9,14 +9,21 @@
 #! format: noindent
 
 # initialize all the values in the container of a general FD block (either straight sided or curved)
-# OBS! Requires the SBP derivative matrix in order to compute metric terms that are free-stream preserving
+# OBS! Requires the SBP derivative matrix in order to compute metric terms. For a standard
+# SBP operator the matrix is passed directly, whereas for an upwind SBP operator we must
+# specifically pass the central differencing matrix.
 function init_element!(elements, element, basis::AbstractDerivativeOperator,
                        corners_or_surface_curves)
     calc_node_coordinates!(elements.node_coordinates, element, get_nodes(basis),
                            corners_or_surface_curves)
 
-    calc_metric_terms!(elements.jacobian_matrix, element, basis,
-                       elements.node_coordinates)
+    if basis isa DerivativeOperator
+        calc_metric_terms!(elements.jacobian_matrix, element, basis,
+                           elements.node_coordinates)
+    else # basis isa SummationByPartsOperators.UpwindOperators
+        calc_metric_terms!(elements.jacobian_matrix, element, basis.central,
+                           elements.node_coordinates)
+    end
```

Comment:
Can we maybe change this as follows to make it a bit more extendable? First, we keep


```suggestion
    calc_metric_terms!(elements.jacobian_matrix, element, basis,
                       elements.node_coordinates)
```

here. Next, we change the stuff below by inserting the specialization

```julia
function calc_metric_terms!(jacobian_matrix, element, D_SBP::SummationByPartsOperators.UpwindOperators,
                            node_coordinates)
    calc_metric_terms!(jacobian_matrix, element, D_SBP.central, node_coordinates)
end
```

after the method for `calc_metric_terms!` that we have below.

The reason for this is that not everything that is a non-upwind operator will be a `DerivativeOperator`, e.g., DG-style methods implemented in SummationByPartsOperators.jl.

Reply:
Yes, this looks better. I only added this `if` statement strategy because I couldn't think of a better / less obtrusive way. I will change this with your suggestion.

Reply:
Done.